### ISSUE TITLE
list flyweight enhancement

### ISSUE THREAD

**Issue by jitsni** [reaktivity/nukleus-maven-plugin#15]
```
    public Builder<B, T> wrap(MutableDirectBuffer buffer, int offset, int maxLimit) {
      super.wrap(buffer, offset, maxLimit);
      super.limit(offset);
-     itemRW.wrap(buffer, offset, maxLimit);
      return this;
    }

    public Builder<B, T> item(Consumer<B> mutator) {
+    itemRW.wrap(buffer(), limit(), maxLimit());
      mutator.accept(itemRW);
      limit(itemRW.build().limit());
-     itemRW.wrap(buffer(), limit(), maxLimit());
      return this;
    }
```

**Comment by cmebarrow** [comment]
@jitsni Could you explain the purpose of these changes?

**Comment by jitsni** [comment]
One less call as we don't need itemRW.wrap() in the list's wrap() method.
Also, the above logic simple, before building an item, you call its builder's wrap() method.

**Comment by cmebarrow** [comment]
PR #30 now implements this enhancement by generating a method <fieldName>Item in the Builder, example: `Builder headersItem(Consumer<HttpHeaderFW.Builder> mutator)`

**Comment by cmebarrow** [comment]
Closed via PR #80
